apps/src/blockly/addons/cdoFieldPicker.js

```diff
@@ -0,0 +1,129 @@
+import GoogleBlockly from 'blockly/core';
+
+const BUTTON_CORNER_RADIUS = 3;
+const BUTTON_INNER_HEIGHT = 16;
+import {DEFAULT_SOUND} from '@cdo/apps/blockly/constants';
+
+/**
+ * This is a customized field which the user clicks to select an option from a customized picker,
+ * for example, the location of a sprite from a grid or a sound file from a customized modal.
+ * @param value The initial value of the field.
+ * @param onChange The function that handles the field's editor.
+ * @param onDisplay The function tht handles how the field text is displayed.
+ * @param buttonIcon If the field displyas a button, this is the string that is displayed on the button.
+ */
+export default class CdoFieldPicker extends GoogleBlockly.Field {
+  constructor(value, onChange, onDisplay, buttonIcon) {
+    super(value);
+    this.onChange = onChange;
+    this.onDisplay = onDisplay;
+    this.buttonIcon = buttonIcon;
+
+    this.SERIALIZABLE = true;
+  }
+
+  static fromJson(options) {
+    return new CdoFieldPicker(
+      options.value,
+      options.onChange,
+      options.onDisplay,
+      options.buttonIcon
+    );
+  }
+
+  /**
+   * Validate the changes to a field's value before they are set.
+   * @override
+   */
+  doClassValidation_(newValue) {
+    if (typeof newValue !== 'string') {
+      return null;
+    }
+    // Handle 'play sound' block with default param from CDO blockly.
+    // TODO: Remove when sprite lab is migrated to Google blockly.
+    if (newValue === 'Choose') {
+      return DEFAULT_SOUND;
+    }
+    return newValue;
+  }
+
+  /**
+   * Create the block UI for this field.
+   * @override
+   */
+  initView() {
+    super.initView();
+    if (this.buttonIcon) {
+      this.buttonElement_ = Blockly.utils.dom.createSvgElement(
```

Comment:
What would happen if you tried to create a field for the sound block and included `true` for `buttonIcon`?

Reply:
If you pass `true` for `buttonIcon`, you receive an error: 'Cannot read properties of null (reading 'setAttribute')'.
If you pass a SVG <tspan> element, then the icon will be displayed on the button.
Example:
```
const buttonIcon = document.createElementNS(SVG_NS, 'tspan');
  buttonIcon.style.fontFamily = 'FontAwesome';
  buttonIcon.textContent = '\u2318';
  return new Blockly.FieldPicker('Choose', onChange, onDisplay, 'sound');
```
![Screen Shot 2023-05-01 at 9 51 03 AM](https://user-images.githubusercontent.com/107423305/235472135-51d14212-0e85-4bde-bb3d-d57bf6e6c7a1.png)

I added a comment in the constructor that `buttonIcon` is a SVG <tspan> element. Thanks for the question!

Reply:
I meant to ask about passing a valid value. Thanks for answering both cases!